src/types/plan.d.ts

```diff
@@ -2,12 +2,19 @@ type Plan = PlanSchema & { end_time_doy: string; start_time_doy: string };
 
 type PlanInsertInput = Pick<PlanSchema, 'duration' | 'model_id' | 'name' | 'start_time'>;
 
+type RelatedPlan = {
+  id: number;
+  name: string;
+};
+
 type PlanSchema = {
+  child_plans: RelatedPlan[];
   duration: string;
   id: number;
   model: Model;
   model_id: number;
   name: string;
+  parent_plan: RelatedPlan | null;
```

Comment:
We should just do`Pick<PlanSchema, 'id' | 'name'>` instead of `RelatedPlan` to keep the sub-types relative to the parent.